auth/common.cc

```diff
@@ -45,15 +45,13 @@ static logging::logger auth_log("auth");
 // Func must support being invoked more than once.
 future<> do_after_system_ready(seastar::abort_source& as, seastar::noncopyable_function<future<>()> func) {
     struct empty_state { };
-    return delay_until_system_ready(as).then([&as, func = std::move(func)] () mutable {
-        return exponential_backoff_retry::do_until_value(1s, 1min, as, [func = std::move(func)] {
-            return func().then_wrapped([] (auto&& f) -> std::optional<empty_state> {
-                if (f.failed()) {
-                    auth_log.debug("Auth task failed with error, rescheduling: {}", f.get_exception());
-                    return { };
-                }
-                return { empty_state() };
-            });
+    return exponential_backoff_retry::do_until_value(1s, 1min, as, [func = std::move(func)] {
```

Comment:
What is the 1s in this loop? I hope it doesn't guarantee we wait 1s (which doesn't sound a lot, but it's longer than the entire single-node Scylla boot...)

Reply:
It would cause a 1s wait **if** the function needs to be retried. For a specific case of single-node clusters, it's almost never the case, because a QUORUM insert is trivially fulfilled.

Reply:
I see. I think maybe 1s is excessive, but given you're saying (and I'll test now ;-)) it usually doesn't happen in normal tests, and this is what the existing code did anyway, I'll accept.